test/data/valid_apache_zookeeper.tf

```diff
@@ -0,0 +1,31 @@
+provider "instaclustr" {
+    username = "%s"
+    api_key = "%s"
+    api_hostname = "%s"
+}
+resource "instaclustr_cluster" "validApacheZookeeper" {
+    cluster_name = "test_zookeeper_cluster"
+    node_size = "zookeeper-developer-t3.small-20"
+    data_centre = "US_WEST_2"
+    sla_tier = "NON_PRODUCTION"
+    cluster_network = "192.168.0.0/18"
+    private_network_cluster = false
+    pci_compliant_cluster = false
+
+    cluster_provider = {
+        name = "AWS_VPC"
+    }
+
+    rack_allocation = {
+        number_of_racks = 3
+        nodes_per_rack = 1
+    }
+    
+    bundle {
+        bundle = "APACHE_ZOOKEEPER"
+        version = "apache-zookeeper:3.5.8"
+        options = {
+            zookeeper_node_count = 3
```

Comment:
**[Question]** This doesn't feel right, since this is a main offering the node count should be calculated through the rack allocation. Is there something different here that I'm missing?

Reply:
So for this bundle it is similar to REDIS in a sense that we don't need the rack allocation. The rack allocation is there because the version I'm PR-ing against still have a problem whereby rack allocation is "expected", i.e., we can create a zookeeper or Redis cluster without rack allocation, but then the cluster read still returns rack allocation, which makes terraform think there is a need for an update where it shouldn't be.

Reply:
Alrighty, that makes sense. Terraform's deadly self-update cycle can be a pain. I suppose it's something that needs to be addressed in future with the structuring of the resources.